crates/bevy_ecs/src/system/function_system.rs

```diff
@@ -140,6 +140,7 @@ pub struct SystemState<Param: SystemParam> {
 
 impl<Param: SystemParam> SystemState<Param> {
     pub fn new(world: &mut World) -> Self {
+        let _ = <Param::Fetch as SystemParamState>::world_access_level();
```

Comment:
Yeah, if `world_access_level()` is called on an invalid tuple of params, it'll panic. If systems can panic on construction, system states can too. Would a comment be sufficient?

Reply:
TIL that `_` and `_panic_on_invalid` behave differently. https://stackoverflow.com/a/65733313. so this should probably keep using `_`.

I'm fine with a comment. The problem is that the function name `world_access_level` isn't clear that it does a check. I understand that the name does make sense since most of the time it is just an getter method and only really does a check when the Param is a tuple of Params. It's probably not worth making a `check_world_access_level` function.

Reply:
I decided to make it a separate function after all. Looks clearer and might come in handy for documentation.